packages/clerk-js/src/utils/__tests__/url.test.ts

```diff
@@ -431,7 +431,22 @@ fdescribe('isAllowedRedirectOrigin', () => {
     ['https://test.clerk.com/foo?hello=1', [/https:\/\/www\.clerk\.com/], false],
   ];
 
+  const warnMock = jest.spyOn(global.console, 'warn').mockImplementation();
+
+  beforeEach(() => warnMock.mockClear());
+  afterAll(() => warnMock.mockRestore());
+
   test.each(cases)('isAllowedRedirectOrigin("%s","%s") === %s', (url, allowedOrigins, expected) => {
     expect(isAllowedRedirectOrigin(url, allowedOrigins)).toEqual(expected);
+
+    if (!expected) {
+      const output = warnMock.mock.calls[0][0];
+
+      expect(warnMock).toHaveBeenCalledTimes(1);
+      expect(output.includes(new URL(url))).toBe(true);
```

Comment:
Must be passed through `new URL` or the format is different from the message.

I separated the url check from the `toMatch` call as `new Regexp(...)` doesn't appropriately handle `?`. This just felt cleaner and simpler.